gensim/similarities/index.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2013 Radim Rehurek <me@radimrehurek.com>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+try:
+    from annoy import AnnoyIndex
+except ImportError:
+    raise ImportError("Annoy has not been installed, if you wish to use the annoy indexer, please run `pip install annoy`")
+
+
+class SimilarityIndex(object):
+
+    @classmethod
+    def build_from_word2vec(cls, model, num_trees):
```

Comment:
We can make specialized wrappers for specific models, but the core API functionality should definitely be an arbitrary matrix/sequence of vectors (like the existing Similarity interfaces).